pkg/build/builder/util.go

```diff
@@ -46,26 +56,83 @@ func readNetClsCGroup(reader io.Reader) string {
 	names := []string{"net_cls", "cpu"}
 	for _, group := range names {
 		if value, ok := cgroups[group]; ok {
-			return value
+			return value, containerType
+		}
+	}
+
+	return "", containerType
+}
+
+// readResolveConfHostPath determines the path to the resolv.conf file for this
+// container, as it exists on the host machine.  (/etc/resolv.conf is mounted
+// into the container from the host path).
+func readResolvConfHostPath() (string, error) {
+	// find the /etc/resolv.conf host path based on what is mounted into this
+	// container.
+	resolvConf, err := os.Open("/proc/1/mountinfo")
+	if err != nil {
+		return "", err
+	}
+	defer resolvConf.Close()
+
+	scanner := bufio.NewScanner(resolvConf)
+	for scanner.Scan() {
+		if match := resolvConfPattern.FindStringSubmatch(scanner.Text()); match != nil {
+			return match[1], nil
 		}
 	}
+	return "", fmt.Errorf("Unable to determine /etc/resolv.conf hostpath")
+}
 
-	return ""
+// readPid determines the actual host pid of the pid 1 process in this container
+func readPid() (string, error) {
+	// get pid from /proc/1/sched , e.g.: "java (8151, #threads: 53)"
+	pidFile, err := os.Open("/proc/1/sched")
```

Comment:
Wow... that's hacky.

Reply:
and, let me emphasize, temporary until we move to pure crio support, however long that takes.

Reply:
Cleaner way for getting pid as well as resolv path is from crio inspect endpoint but we didn't want that.. and yes this doesn't work with pid namespaces but hopefully we can migrate away from docker for builds by then or switch to using crio inspect endpoint.